File: code/backend/foodometer-item-service/src/main/java/com/springmicro/foodometer/document/FoodItem.java
package com.springmicro.foodometer.document;

import com.springmicro.foodometer.constants.FoodItemCategoryEnum;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Data
@Document(collection = "FoodItem")
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class FoodItem {

    @Id
    private String id;
    private FoodItemCategoryEnum category;
    private String itemName;
    private Double itemPrice;
}

File: code/frontend/food-o-gui/src/app/_model/food-item.ts
export class FoodItem {
    id: string;
    category: string;
    itemName: string;
    itemPrice: number;


    constructor(
        id?: string,
        category?: string,
        itemName?: string,
        itemPrice?: number
    ) {
        this.id = id;
        this.category = category!;
        this.itemName = itemName!;
        this.itemPrice = itemPrice!;
    }
}
